#VRML_SIM R2021b utf8
WorldInfo {
  basicTimeStep 128
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation -0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
    ContactProperties {
      material1 "ExteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation 0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
  ]
}
Viewpoint {
  orientation 0.47756485767984364 -0.8550282212741263 -0.20213497355488685 5.394974210561848
  position 23.348368305823016 13.721567723321085 18.167060499988693
}
TexturedBackground {
}
TexturedBackgroundLight {
  castShadows FALSE
}
RectangleArena {
  floorSize 25 25
  floorTileSize 50 50
}
DEF Youbot Youbot {
  translation 0 0.085297 0
  rotation 0.999987208574567 -0.0013813423517076782 0.004865653147609176 -1.5710938681740114
  controller "validate_inputs"
  supervisor TRUE
  bodySlot [
    Accelerometer {
    }
    GPS {
    }
    Compass {
    }
    DEF ForwardLowResBigFov Camera {
      translation 0.3 0 0
      rotation 0.5773506025225371 -0.5773506025225371 -0.5773496025232256 2.09439
      name "ForwardLowResBigFov"
      fieldOfView 1
      width 128
    }
    DEF ForwardHighResSmallFov Camera {
      translation 0.3 0 0
      rotation 0.5773506025225371 -0.5773506025225371 -0.5773496025232256 2.09439
      name "ForwardHighResSmallFov"
      fieldOfView 0.5
      width 256
      height 128
    }
    DEF ForwardHighResBigFov Camera {
      translation 0.3 0 0
      rotation 0.5773506025225371 -0.5773506025225371 -0.5773496025232256 2.09439
      name "ForwardHighRes"
      fieldOfView 1
      width 256
      height 128
    }
    DEF ForwardHighResSmallFov Camera {
      translation 0.3 0 0
      rotation 0.5773506025225371 -0.5773506025225371 -0.5773496025232256 2.09439
      name "ForwardHighResSmall"
      fieldOfView 0.5
      width 256
      height 128
    }
    DEF BackLowRes Camera {
      translation -0.27 -2.00826e-06 1.02099e-06
      rotation 0.5773486025216047 0.5773526025188509 0.5773496025209163 2.0944
      name "BackLowRes"
      fieldOfView 1
      width 128
    }
    DEF RightLowRes Camera {
      translation -0.00999658 -0.200002 1.57252e-06
      rotation -6.106782428407177e-07 0.7071072811864377 0.70710628118604 -3.1415853071795863
      name "RightLowRes"
      fieldOfView 1
      width 128
    }
    DEF LeftLowRes Camera {
      translation -0.00999894 0.189998 -2.08147e-07
      rotation -0.9999999999879128 -4.8411699999414835e-06 -8.588339999896191e-07 -1.5707953071795862
      name "LeftLowRes"
      fieldOfView 1
      width 128
    }
    DEF BackHighRes Camera {
      translation -0.27 -2.00826e-06 1.02099e-06
      rotation 0.5773486025216047 0.5773526025188509 0.5773496025209163 2.0944
      name "BackHighRes"
      fieldOfView 1
      width 256
      height 128
    }
    Gyro {
    }
    LightSensor {
    }
    Receiver {
    }
    RangeFinder {
      translation 0.41 0 0
      rotation -0.5773502691896258 0.5773502691896258 0.5773502691896258 -2.0944
      maxRange 5
    }
    Lidar {
      rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
      numberOfLayers 1
      near 0.3
      minRange 0.3
      maxRange 10
      type "rotating"
    }
    Group {
    }
  ]
}
DEF Berry0 Solid {
  translation 0.23022346837427063 0.1966107787459715 -11.420111734187133
  rotation -0.4472135954998153 2.4166820336214355e-12 -0.8944271909999872 0.0010477170011384145
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.2 0.1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "solid(6)"
  boundingObject DEF BOX Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
}
DEF Berry1 Solid {
  translation 8.31007 0.0444805 -11.4201
  rotation -0.4472135954998153 2.4166820336214355e-12 -0.8944271909999872 0.0010477170011384145
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.2 0.1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject DEF BOX Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
}
DEF Berry2 Solid {
  translation 10.0601 0.0433653 -10.3001
  rotation -0.4472135954998153 2.4166820336214355e-12 -0.8944271909999872 0.0010477170011384145
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.2 0.1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "solid(2)"
  boundingObject DEF BOX Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
}
DEF Berry3 Solid {
  translation 1.14007 0.0603458 8.0999
  rotation -0.4472135954998153 2.4166820336214355e-12 -0.8944271909999872 0.0010477170011384145
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.2 0.1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "solid(3)"
  boundingObject DEF BOX Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
}
DEF zombie0 Pedestrian {
  translation -7.707573600054271 1.27 -7.145606399945701
  rotation 0 1 0 0.8080057916952204
  name "pedestrian1"
  controller "blue_zombie"
  controllerArgs [
    "--trajectory=\"\""
    "--speed=1.5"
  ]
  shirtColor 0 0.5 1
  pantsColor 0 0.5 1
  shoesColor 0 0.5 1
  skinColor 0 0.5 1
  bodySlot [
    Emitter {
      range 4
      channel -1
    }
  ]
  rightLegAngle 0.23563636363636417
  rightLowerLegAngle 0.13490909090909156
  rightFootAngle 0.40472727272727466
  leftLegAngle -1.1294545454545468
  leftLowerLegAngle 1.705272727272728
  leftFootAngle -0.06490909090909155
  rightArmAngle -0.34072727272727515
  rightLowerArmAngle -0.5698181818181831
  rightHandAngle -0.24709090909090947
  leftArmAngle 0.5534545454545489
  leftLowerArmAngle -0.6803636363636388
  leftHandAngle 0.11563636363636419
  headAngle 0.003272727272726854
}
DEF zombie1 Pedestrian {
  translation -7.62613 1.27 9.10663
  rotation 0 1 0 2.4944603805566383
  name "pedestrian(1)"
  controller "green_zombie"
  controllerArgs [
    "--trajectory=\"\""
    " --speed=1.5"
  ]
  shirtColor 0 0.7 0
  pantsColor 0 0.7 0
  shoesColor 0 0.7 0
  skinColor 0 0.7 0
  bodySlot [
    Emitter {
      range 4
      channel -1
    }
  ]
  leftHandSlot [
    Emitter {
    }
  ]
  rightLegAngle 0.23563636363636417
  rightLowerLegAngle 0.13490909090909156
  rightFootAngle 0.40472727272727466
  leftLegAngle -1.1294545454545468
  leftLowerLegAngle 1.705272727272728
  leftFootAngle -0.06490909090909155
  rightArmAngle -0.34072727272727515
  rightLowerArmAngle -0.5698181818181831
  rightHandAngle -0.24709090909090947
  leftArmAngle 0.5534545454545489
  leftLowerArmAngle -0.6803636363636388
  leftHandAngle 0.11563636363636419
  headAngle 0.003272727272726854
}
DEF zombie4 Pedestrian {
  translation -0.2575249436194565 1.27 6.951264149888269
  rotation 0 1 0 -2.4126957399490436
  name "pedestrian(5)"
  controller "purple_zombie"
  controllerArgs [
    "--trajectory=\"\""
    " --speed=1.5"
  ]
  shirtColor 0.6 0.2 1
  pantsColor 0.6 0.2 1
  shoesColor 0.6 0.2 1
  skinColor 0.6 0.2 1
  bodySlot [
    Emitter {
      range 4
      channel -1
    }
  ]
  rightLegAngle 0.23563636363636417
  rightLowerLegAngle 0.13490909090909156
  rightFootAngle 0.40472727272727466
  leftLegAngle -1.1294545454545468
  leftLowerLegAngle 1.705272727272728
  leftFootAngle -0.06490909090909155
  rightArmAngle -0.34072727272727515
  rightLowerArmAngle -0.5698181818181831
  rightHandAngle -0.24709090909090947
  leftArmAngle 0.5534545454545489
  leftLowerArmAngle -0.6803636363636388
  leftHandAngle 0.11563636363636419
  headAngle 0.003272727272726854
}
DEF zombie5 Pedestrian {
  translation 0.00222015 1.27 -7.98276
  rotation 0 1 0 2.542269011900159
  name "pedestrian(4)"
  controller "acqua_zombie"
  controllerArgs [
    "--trajectory=\"\""
    " --speed=1.5"
  ]
  shirtColor 0 0.9 0.7
  pantsColor 0 0.9 0.7
  shoesColor 0 0.9 0.7
  skinColor 0 0.9 0.7
  bodySlot [
    Emitter {
      range 4
      channel -1
    }
  ]
  rightLegAngle 0.23563636363636417
  rightLowerLegAngle 0.13490909090909156
  rightFootAngle 0.40472727272727466
  leftLegAngle -1.1294545454545468
  leftLowerLegAngle 1.705272727272728
  leftFootAngle -0.06490909090909155
  rightArmAngle -0.34072727272727515
  rightLowerArmAngle -0.5698181818181831
  rightHandAngle -0.24709090909090947
  leftArmAngle 0.5534545454545489
  leftLowerArmAngle -0.6803636363636388
  leftHandAngle 0.11563636363636419
  headAngle 0.003272727272726854
}
